turbopack/crates/turbo-rcstr/src/lib.rs

```diff
@@ -44,13 +56,41 @@ use turbo_tasks_hash::{DeterministicHash, DeterministicHasher};
 // If you want to change the underlying string type to `Arc<str>`, please ensure that you profile
 // performance. The current implementation offers very cheap `String -> RcStr -> String`, meaning we
 // only pay for the allocation for `Arc` when we pass `format!("").into()` to a function.
-#[derive(Clone, Default, PartialEq, Eq, PartialOrd, Ord, Hash, Serialize, Deserialize)]
-#[serde(transparent)]
-pub struct RcStr(Arc<String>);
+pub struct RcStr {
+    unsafe_data: TaggedValue,
+}
+
+unsafe impl Send for RcStr {}
+unsafe impl Sync for RcStr {}
+
+const DYNAMIC_TAG: u8 = 0b_00;
+const INLINE_TAG: u8 = 0b_01; // len in upper nybble
+const INLINE_TAG_INIT: NonZeroU8 = unsafe { NonZeroU8::new_unchecked(INLINE_TAG) };
+const TAG_MASK: u8 = 0b_11;
+const LEN_OFFSET: usize = 4;
+const LEN_MASK: u8 = 0xf0;
 
 impl RcStr {
+    pub fn new(s: Cow<str>) -> Self {
+        new_atom(s)
+    }
+
```

Comment:
`new` makes sense if you have a preferred constructor. If somebody already has a `String` (the common case) and decides to wrap it in a `Cow` in order to use this constructor (which would show up at the top of the docs), they'd have worse code than if they'd just called `.into()`.

For that reason, I think we should get rid of this explicit constructor and stick to the advice (in the documentation above on `RcStr` of using `.into()` or `RcStr::from`.